internal/images/images.go

```diff
@@ -397,22 +398,27 @@ func parseLeadingNumber(s string) (v float64, ok bool) {
 	i := 0
 
 	for _, r := range s {
+		if r >= utf8.RuneSelf {
+			break
+		}
+
 		if unicode.IsDigit(r) || r == '+' || r == '-' || r == '.' {
 			i++
 		} else {
 			break
 		}
 	}
+
 	if i == 0 {
 		return 0, false
 	}
 	n, err := strconv.ParseFloat(s[:i], 64)
 	return n, err == nil
 }
 
-// dimsFromSVG reads tokens until it sees the first <svg>
-// start-element, then extracts width/height/viewBox.
-func dimsFromSVG(r io.Reader) (w, h float64, err error) {
+// dimsFromSVG reads XML tokens from SVG file r until it can compute
+// the dimensions of the image (in "abstract user units").
+func dimsFromSVG(r io.Reader) (float64, float64, error) {
```

Comment:
I would still name these results for documentation purposes.

It's okay to have the `w, ok :=` and `h, ok :=` variables below be different variables (with the same name in a shadowed scope)